Fails to build on ARMEL

**Describe your environment** 
Debian unstable on armel (32-bit little-endian ARM architecture using the Embedded ABI - v5te instruction set)

**Steps to reproduce**
Build opentelemetry-cpp on armel.

**What is the expected behavior?**
Successful build.

**What is the actual behavior?**

The build fails because of:

```
[  8%] Building CXX object api/test/trace/CMakeFiles/api_provider_test.dir/provider_test.cc.o
cd /home/santiago/opentelemetry-cpp-1.17.0/obj-arm-linux-gnueabi/api/test/trace && /usr/bin/c++ -DHAVE_ABSEIL -DOPENTELEMETRY_ABI_VERSION_NO=1 -I/home/santiago/opentelemetry-cpp-1.17.0/obj-arm-linux-gnueabi/generated/third_party/opentelemetry-proto -I/home/santiago/opentelemetry-cpp-1.17.0/api/include -g -O2 -ffile-prefix-map=/home/santiago/opentelemetry-cpp-1.17.0=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -Wdate-time -D_FORTIFY_SOURCE=2 -Wno-error=deprecated-declarations -DGTEST_HAS_PTHREAD=1 -MD -MT api/test/trace/CMakeFiles/api_provider_test.dir/provider_test.cc.o -MF CMakeFiles/api_provider_test.dir/provider_test.cc.o.d -o CMakeFiles/api_provider_test.dir/provider_test.cc.o -c /home/santiago/opentelemetry-cpp-1.17.0/api/test/trace/provider_test.cc 
/tmp/cc6SyTzU.s: Assembler messages:
/tmp/cc6SyTzU.s:18172: Error: selected processor does not support `yield' in ARM mode
/tmp/cc6SyTzU.s:22037: Error: selected processor does not support `yield' in ARM mode
make[3]: *** [api/test/trace/CMakeFiles/api_provider_test.dir/build.make:79: api/test/trace/CMakeFiles/api_provider_test.dir/provider_test.cc.o] Error 1
```

You can see a build full log (with parallel jobs) at: https://buildd.debian.org/status/fetch.php?pkg=opentelemetry-cpp&arch=armel&ver=1.17.0-1&stamp=1730821961&raw=0

**Additional context**
N/A

I will propose a fix for this.

--- a/api/include/opentelemetry/common/spin_lock_mutex.h
+++ b/api/include/opentelemetry/common/spin_lock_mutex.h
@@ -68,8 +68,10 @@ public:
 #  else
     __builtin_ia32_pause();
 #  endif
-#elif defined(__arm__)
-    __asm__ volatile("yield" ::: "memory");
+#elif defined(__armel__) || defined(__ARMEL__)
+    asm volatile("nop" ::: "memory");
+#elif defined(__arm__) || defined(__aarch64__)  // arm big endian / arm64
+    __asm__ __volatile__("yield" ::: "memory");
 #else
     // TODO: Issue PAGE/YIELD on other architectures.
 #endif

--- a/api/test/common/spinlock_benchmark.cc
+++ b/api/test/common/spinlock_benchmark.cc
@@ -102,8 +102,10 @@ static void BM_ProcYieldSpinLockThrashing(benchmark::State &s)
 #  else
           __builtin_ia32_pause();
 #  endif
-#elif defined(__arm__)
-          __asm__ volatile("yield" ::: "memory");
+#elif defined(__armel__) || defined(__ARMEL__)
+          asm volatile("nop" ::: "memory");
+#elif defined(__arm__) || defined(__aarch64__)  // arm big endian / arm64
+          __asm__ __volatile__("yield" ::: "memory");
 #endif
         }
       },

